### ISSUE TITLE
New-AzADServicePrincipal creating role and scope assignment by default

### ISSUE THREAD

**Issue by bgawale** [Azure/azure-powershell#10963]
## Description

As per the documentation of the [New-AzADServicePrincipal](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzADServicePrincipal.md) it is mentioned that - 

> This cmdlet has the ability to assign a role to the service principal with the Role and Scope parameters; if neither of these parameters are provided, no role will be assigned to the service principal. The default values for the Role and Scope parameters are "Contributor" and the current subscription, respectively (note: the defaults are only used when the user provides a value for one of the two parameters, but not the other).

However it seems that it is not working as expected when new service principal is created with parameter below

## Steps to reproduce
Create a new azure ad app
```powershell
$Application = New-AzADApplication -IdentifierUris "some_uri" -DisplayName "some_name"
```
and then try generating the service principal for the new ad app without specifying role as well as the scope parameter
```powershell
 $ServicePrincipal = New-AzADServicePrincipal -ApplicationId $Application.ApplicationId
```
Since role and scope both parameters are not passed, so as per the documentation above It is expected that scope and role should not be set for the newly created service principal, however it is still assigning subscription level scope with contributor role to the service principal. 
## Environment data

```
Name                           Value
----                           -----
PSVersion                      5.1.18362.145
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.145
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Module versions

```powershell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.9.1      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment...}
```

## Debug output

```

```

## Error output

```

```

**Comment by markcowl** [comment]
@bgawale Documentation needs to be updated to reflect the intended behavior, which is that a contributor role is added to the subscription (if the user has sufficient privileges).

**Comment by bgawale** [comment]
I guess its not really the documentation change and it appears to be correct. This is based on the fact that the behaviour of the relevant AzureRm PS command i.e. New-AzureRmADServicePrincipal. 
It does not assign any role and scope to the service principal when scope and role parameters are not passed and same was expected from the Az command. 

Example

```
$Application = New-AzureRmADApplication -IdentifierUris "some_uri" -DisplayName "some_name"
$ServicePrincipal = New-AzureRmADServicePrincipal -ApplicationObject $Application

```
Note that the code above does not pass any role and scope parameters while creating the service principal and as correctly expected, it does not by default assign any scope and role to the service principal. Why can't we expect same behavior from the Az command?

**Comment by bgawale** [comment]
Thoughts on the comment above?

**Comment by MertSenel** [comment]
you can use method below as workaround
```
$Application = New-AzADApplication -IdentifierUris "some_uri" -DisplayName "some_name"
$ServicePrincipal = New-AzADServicePrincipal -ApplicationObject $Application

#Remove any role assignments that are being created by the New-AzADServicePrincipal 
Get-AzRoleAssignment -ObjectId $ServicePrincipal.Id | % { $_ | Remove-AzRoleAssignment }
```

**Comment by wahyuen** [comment]
Is the intention here that the team is leaving as 'current functionality' and updating doco? Or is the doco correct and they are correcting the functionality? 

One of our customers just asked (and quite rightly) why we had extra permissions at a subscription level even though our scripts don't explicitly call out assignment to it and we are left trying to explain that this 'maybe' a bug or we 'might' have to build workarounds....

**Comment by dcaro** [comment]
@bgawale @wahyuen thanks for reporting the issue and I'm sorry that it took so long but be assured that we are now actively working on fixing this.

**Comment by wahyuen** [comment]
so instead of 'fixing' the issue, you choose to update documentation instead.....would of been nice had this been communicated as previously asked multiple times >_>

**Comment by bgawale** [comment]
Can someone help us understand how this documentation fix should be called as 'fix' when Rm equivalent of same command functions differently? 
If the documentation is the fix for this az command, then does this mean Rm equivalent of same command has a bug?

**Comment by dcaro** [comment]
When the Azure PowerShell modules switched from the AzureRM generation to Az, we aimed at having the default behavior of the cmdlets aligned with the most common scenarios.
A common scenario when creating a service principal is to have the contributor role, so when we we created the Az generation we changed the default behavior in `New-AzADServicePrincipal` to assign the contributor role. The intended behavior of `New-AzureRMAdServicePrincipal` was to not do the role assignment.
The issue was that this part of the documentation had not been updated to reflect the expected behavior. I'm sorry if this has been misleading. 

In summary the expected behavior when the `role` and `scope` parameters are not set is as follow:
`New-AzAdServicePrincipal` -> contributor role is assigned to the service principal
`New-AzureRMAdServicePrincipal` -> not role is assigned to the service principal
